src/websocket.js

```diff
@@ -434,10 +434,14 @@ Strophe.Websocket = class Websocket {
                     } else {
                         stanza = data[i];
                     }
-                    const rawStanza = Strophe.serialize(stanza);
-                    this._conn.xmlOutput(stanza);
-                    this._conn.rawOutput(rawStanza);
-                    this.socket.send(rawStanza);
+                    try {
+                      const rawStanza = Strophe.serialize(stanza);
+                      this._conn.xmlOutput(stanza);
+                      this._conn.rawOutput(rawStanza);
+                      this.socket.send(rawStanza);
+                    } catch (error) {
+                      Strophe.error("Websocket could not send stanza, error: " + JSON.stringify(error));
```

Comment:
If you catch the error here and just log it, then on line 447 `this._conn._data` is cleared.
This means that the stanzas that were queued to be sent are now gone and will never be sent out.

Do you know why it happens that you suddenly have no socket anymore even though there still are stanzas to be sent out? Seems to me like there might be something buggy in your code that causes this situation and that the problem might better be fixed there instead?

Also, concerning `try/catch` clauses, it's good practice to put the clause only around the exact line(s) of code that could reasonably cause an exception. Lines which should never cause an exception, like `const rawStanza = Strophe.serialize(stanza);` should not be wrapped inside a `try/catch`.

Reply:
Thank You for your comments, @jcbrand 
I didn't investigate the reason deep in Strophe that time.
I hope I have another iteration about it soon.

I can describe our setup to provide a context and listen for ideas on what could be done better.
We are using Strophe in browser and mobile.

We implemented ping/pong mechanism and network timeout request.
But the error ^^ happened before that.

For browser If we put the network in big latency conditions and try to send requests the error popups after some time.
![image](https://user-images.githubusercontent.com/20629939/144594861-bd463236-0238-48c3-a534-4b623517c691.png)

Current logic we have: If we don't have pong back in networkTimeout then we disconnect and connect again.
I suspect that it could happen while we disconnected we are trying to send ping(or another query).

On Mobile, it's simpler. As soon as the app goes to the background WS is killed on iOS and lives up to 2 mins and is killed after on Android.
Even we didn't have ping/pong the error happened. So indeed we are trying to send smth while WS is dead.

Thanks for possibility to explain and understand better :)
I will convert it into draft and come back to the investigation when I have a chance.

Reply:
It turned out that the disconnect_timeout I use works synchronously 😛
I was too tied on Strophe disconnection flow, by setting it to 0. :)
I put 50ms to give time for presence unavailable and WS close for being sent.

I'll keep the PR in draft for the next two weeks in case we discover smth else.

Reply:
Ok, thanks for writing down your findings here.